Plutarch/Repr/Data.hs

```diff
@@ -0,0 +1,209 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Plutarch.Repr.Data (
+  PDataStruct (PDataStruct, unPDataStruct),
+  PDataRec (PDataRec, unPDataRec),
+  DeriveAsDataRec (DeriveAsDataRec, unDeriveAsDataRec),
+  DeriveAsDataStruct (DeriveAsDataStruct, unDeriveAsDataStruct),
+) where
+
+import Data.Kind (Type)
+import GHC.Exts (Any)
+import Generics.SOP
```

Comment:
We already have a qualified import - is this for types? If so, should be explicit.